lib/src/features/game/domain/board_state.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:meta/meta.dart';
+import 'package:dartchess/dartchess.dart';
+
+import './game_status.dart';
+
+@immutable
+class BoardState {
+  final int whiteTime;
+  final int blackTime;
+  final GameStatus status;
+  final List<String> uciMoves;
+  final List<String> sanMoves;
+  final Position<Chess> position;
+
+  const BoardState(
+      {required this.whiteTime,
+      required this.blackTime,
+      required this.status,
+      required this.uciMoves,
+      required this.sanMoves,
+      required this.position});
+
+  factory BoardState.fromJson(Map<String, dynamic> json) {
+    final uciMoves =
+        json['moves'].split(' ').where((m) => m.isNotEmpty).toList();
+    Position<Chess> pos = Chess.initial;
+    List<String> sanMoves = [];
+    for (final m in uciMoves) {
+      final move = Move.fromUci(m);
+      final newPos = pos.playToSan(move);
+      pos = newPos.item1;
+      sanMoves.add(newPos.item2);
+    }
+    return BoardState(
+      whiteTime: json['wtime'],
+      blackTime: json['btime'],
+      status: GameStatus.values.firstWhere((e) => e.name == json['status'],
+          orElse: () => GameStatus.unknown),
+      uciMoves: List.unmodifiable(uciMoves),
+      sanMoves: List.unmodifiable(sanMoves),
+      position: pos,
+    );
+  }
+
+  Move? get lastMove =>
+      uciMoves.isNotEmpty ? Move.fromUci(uciMoves[uciMoves.length - 1]) : null;
+  bool get abortable => status == GameStatus.started && position.fullmoves < 1;
+  bool get resignable => status == GameStatus.started && position.fullmoves > 1;
+  bool get playing => status == GameStatus.started;
+  bool get isLastMoveCapture {
+    final lm = sanMoves.isNotEmpty ? sanMoves[sanMoves.length - 1] : null;
+    return lm != null ? lm.contains('x') : false;
+  }
+
+  @override
+  bool operator ==(Object other) {
+    return other is BoardState &&
+        other.whiteTime == whiteTime &&
+        other.blackTime == blackTime &&
+        other.status == status &&
+        other.uciMoves == uciMoves &&
+        other.sanMoves == sanMoves &&
+        other.position == position;
+  }
```

Comment:
do we need this when we already have `hashCode`?

Reply:
Looks like we don't indeed. But it's the standard way of overriding `==` I've seen it everywhere.

But I won't leave that code, I'll refactor it and use instead https://pub.dev/packages/equatable (or freezed where it makes sense).
In dartchess too.